aten/src/ATen/native/cpu/FusedAdamKernel.cpp

```diff
@@ -238,13 +238,19 @@ std::enable_if_t<
         param_ptr[d] = param_ptr[d] * scalar_t(1 - lr * weight_decay);
       }
     }
-    // exp_avg.lerp_(grad, 1 - beta1)
-    auto is_lerp_weight_small = std::abs(scalar_t(exp_avg_grad_coefficient)) < scalar_t(0.5);
-    if (is_lerp_weight_small) {
-      exp_avg_ptr[d] = exp_avg_ptr[d] + scalar_t(exp_avg_grad_coefficient) * (grad_val - exp_avg_ptr[d]);
-    } else {
-      exp_avg_ptr[d] = grad_val - (grad_val - exp_avg_ptr[d]) * (scalar_t(1) - scalar_t(exp_avg_grad_coefficient));
+    // exp_avg.lerp_(grad, 1 - beta1) if beta1 > 0 else grad  
+    if (exp_avg_grad_coefficient < 1){
+      auto is_lerp_weight_small = std::abs(scalar_t(exp_avg_grad_coefficient)) < scalar_t(0.5);
+      if (is_lerp_weight_small) {
+        exp_avg_ptr[d] = exp_avg_ptr[d] + scalar_t(exp_avg_grad_coefficient) * (grad_val - exp_avg_ptr[d]);
+      } else {
+        exp_avg_ptr[d] = grad_val - (grad_val - exp_avg_ptr[d]) * (scalar_t(1) - scalar_t(exp_avg_grad_coefficient));
+      }
+    }
+    else{
+      exp_avg_ptr[d] = grad_ptr[d];
```

Comment:
`else exp_avg = grad` because `adam_math` is used by AdamW so I added it to keep compatibility.

Reply:
Is this a no-op tho? If this effectively does 

```suggestion
      grad_ptr[d] = grad_ptr[d];
```

Then I would just leave a comment and get rid of the else block as it would be doing unnecessary work

Reply:
It maintains compatibility with AdamW when beta1=0.

Reply:
Ah, would you be okay with changing this PR to update both then? We recently landed a PR that combined AdamW into Adam so it should not be too much more code: https://github.com/pytorch/pytorch/pull/143710